src/components/FilmsList/FilmsList.js

```diff
@@ -1,29 +1,37 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import './FilmsList.scss';
 import { FilmCard } from '../FilmCard';
+import { store } from '../../store';
 
-export const FilmsList = (props) => {
-  const { films } = props;
+export class FilmsList extends React.Component {
+  state = {
+    films: store.getState().films,
+  }
 
-  return (
-    <div className="films">
-      {films.map(film => (
-        <FilmCard key={film.id} {...film} />
-      ))}
-    </div>
-  );
-};
+  unsubscribe = null
 
-FilmsList.propTypes = {
-  films: PropTypes.arrayOf(PropTypes.shape({
-    title: PropTypes.string.isRequired,
-    description: PropTypes.string,
-    imgUrl: PropTypes.string.isRequired,
-    imdbUrl: PropTypes.string.isRequired,
-  })),
-};
+  componentDidMount() {
+    this.unsubscribe = store
+      .subscribe(() => {
+        this.setState({
+          films: store.getState().films,
+        });
+      });
+  }
 
-FilmsList.defaultProps = {
-  films: [],
-};
+  componentWillUnmount() {
+    this.unsubscribe();
+  }
+
+  render() {
+    const { films } = this.state;
+
+    return (
+      <div className="films">
+        {films.map(film => (
+          <FilmCard key={film.id} {...film} />
```

Comment:
if you add two the same movies, key won't be unique

Reply:
I don`t have any idea how to fix it.

Reply:
just generate key with some external tools

Reply:
still the same